File: typescript/6kyu/Persistent_Bugger.ts
export const persistence = (num: number): number => {
  if (num.toString().length === 1) return 0;
  let calc = num;
  let result = 0;
  let numArr = String(num)
    .split("")
    .map((num) => Number(num));
  while (calc.toString().length > 1) {
    calc = numArr.reduce((a, b) => a * b);
    numArr = String(calc)
      .split("")
      .map((num) => Number(num));
    result++;
  }
  return result;
};

File: python/6kyu/Persistent_Bugger.py
def persistence(n):
    if len(str(n)) == 0: 
        return 0
    calc = n
    result = 0
    num_list = list(map(int, str(n)))

    while int(len(str(calc)) > 1):
        calc = 1
        for el in num_list:
            calc = calc * el
        num_list = list(map(int, str(calc)))
        result += 1

    return result